pkg/services/notification/handler.go

```diff
@@ -75,82 +79,205 @@ func (s *Server) CreateNotification(ctx context.Context, req *pb.CreateNotificat
 	if err != nil {
 		return nil, err
 	}
+	logger.Debugf(ctx, "Create notification[%s] in DB successfully.", notification.NotificationId)
 
-	tasks, err := SplitNotificationIntoTasks(notification)
+	_, err = s.createTasksByNotification(ctx, notification)
 	if err != nil {
-		logger.Errorf(ctx, "Split notification into tasks failed, [%+v]", err)
+		logger.Errorf(ctx, "Create tasks by notification failed, [%+v].", err)
 		return nil, err
 	}
 
+	// Enqueue notification after create tasks.
+	err = s.controller.notificationQueue.Enqueue(notification.NotificationId)
+	if err != nil {
+		logger.Errorf(ctx, "Push notification[%s] into etcd failed, [%+v].", notification.NotificationId, err)
+		return nil, err
+	}
+	logger.Debugf(ctx, "Push notification[%s] into etcd successfully.", notification.NotificationId)
+	logger.Debugf(ctx, "Create notification[%s] successfully.", notification.NotificationId)
+
+	return &pb.CreateNotificationResponse{
+		NotificationId: pbutil.ToProtoString(notification.NotificationId),
+	}, nil
+}
+
+func (s *Server) createTasksByNotification(ctx context.Context, nf *models.Notification) ([]*models.Task, error) {
+	tasks, err := rs.SplitNotificationIntoTasks(ctx, nf)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to split notification into tasks, [%+v].", err)
+		return nil, err
+	}
+
+	tasks, err = s.createTasks(ctx, tasks)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to create tasks, [%+v].", err)
+		return nil, err
+	}
+
+	return tasks, nil
+}
+
+func (s *Server) createTasks(ctx context.Context, tasks []*models.Task) ([]*models.Task, error) {
```

Comment:
[]*models.Task in ret is useful?

Reply:
I don't use the return value when I use it. But maybe it need return value in future, so I keep it.
If you insist to remove it ,I can also remove it.

Reply:
done